firmware-binaries/test-cases/switch_calendar_test/src/main.rs

```diff
@@ -0,0 +1,165 @@
+#![no_std]
+#![cfg_attr(not(test), no_main)]
+#![allow(const_item_mutation)]
+#![allow(clippy::empty_loop)]
+#![allow(clippy::approx_constant)]
+
+use bittide_hal::manual_additions::index::IndexTy;
+use bittide_hal::switch_c::ValidEntry;
+use bittide_hal::Index;
+// SPDX-FileCopyrightText: 2025 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+use ufmt::uwrite;
+
+use bittide_hal::hals::switch_c as hal;
+use core::fmt::Write;
+#[cfg(not(test))]
+use riscv_rt::entry;
+
+const INSTANCES: hal::DeviceInstances = unsafe { hal::DeviceInstances::new() };
+
+fn test_result(result: &str) -> ! {
+    let uart = &mut INSTANCES.uart;
+    writeln!(uart, "RESULT: {}", result).unwrap();
+    loop {}
+}
+
+fn test_ok() -> ! {
+    test_result("OK")
+}
+
+fn test_fail(msg: &str) -> ! {
+    let mut full_msg = heapless::String::<1024>::new();
+    let _ = write!(full_msg, "FAIL: {}", msg);
+    test_result(&full_msg)
+}
+
+fn expect<T: core::fmt::Debug + PartialEq>(msg: &str, expected: T, actual: T) {
+    if expected != actual {
+        let mut err = heapless::String::<1024>::new();
+        let _ = write!(err, "{}: expected {:?}, got {:?}", msg, expected, actual);
+        test_fail(&err);
+
+        // Uncomment below to print progress messages
+        // } else {
+        //     writeln!(INSTANCES.uart, "PASS: {} | {:?}", msg, expected).unwrap();
+    }
+}
+
+#[cfg_attr(not(test), entry)]
+fn main() -> ! {
+    let active_entry0: [Index![17]; 16] =
+        core::array::from_fn(|i| unsafe { IndexTy::new_unchecked(i as u8) });
```

Comment:
Is the performance that much worse if you used `index!(i, n = 17)` instead? It's a test case anyway, so doing extra checks shouldn't be a problem.

Same below when IndexTy is used (it shouldn't really be used, only `Index![]` and `index!()`)